sources/TTYIN

```diff
@@ -5600,12 +5600,10 @@ Copyright (c) 1982-1988, 1990-1991, 2021 by Venue & Xerox Corporation.
 
 (TTYIN.SCRATCHFILE
   [LAMBDA NIL
-    (DECLARE (GLOBALVARS TTYINEDIT.SCRATCH))         (* lmm "14-Nov-86 17:05")
-    [COND
-       ([OR (NOT TTYINEDIT.SCRATCH)
-            (NOT (OPENP TTYINEDIT.SCRATCH 'BOTH]
-        (SETQ TTYINEDIT.SCRATCH (OPENSTREAM '{NODIRCORE} 'BOTH 'OLD/NEW NIL
-                                       (CONSTANT (LIST (LIST 'ENDOFSTREAMOP (FUNCTION \TTYIN.RPEOF]
+    (DECLARE (GLOBALVARS TTYINEDIT.SCRATCH))        (* ; "Edited 13-Aug-2021 11:07 by rmk:")
+    (CL:UNLESS (AND TTYINEDIT.SCRATCH (OPENP TTYINEDIT.SCRATCH 'BOTH))
+        [SETQ TTYINEDIT.SCRATCH (OPENSTREAM '{NODIRCORE} 'BOTH 'OLD/NEW
+                                       (CONSTANT (LIST (LIST 'ENDOFSTREAMOP (FUNCTION \TTYIN.RPEOF])
```

Comment:
It took me a long time to see that the difference was the elision of the extra NIL parameter in the OPENSTREAM, which I noticed just as Ron answered.   That's the comment that should have been in the commit for this ("extraneous NIL in OPENSTREAM arguments results in ENDOFSTREAMOP being ignored.")
I'm also not a big fan of using CL functions in code that is otherwise pure Interlisp.